src/checkers/inference/InferenceAnnotatedTypeFactory.java

```diff
@@ -588,5 +596,18 @@ protected InferenceViewpointAdapter createViewpointAdapter() {
     public AnnotatedTypeMirror getTypeOfExtendsImplements(Tree clause) {
         return getAnnotatedTypeFromTypeTree(clause);
     }
+
+    @Override
```

Comment:
To get the  the annotated type of an anonymous class with an explicit annotation, convert the explicit annotation to @VarAnnot

Reply:
Could you also turn this into a code comment?